nautobot/extras/api/views.py

```diff
@@ -441,9 +441,18 @@ def sync(self, request, pk):
             raise CeleryWorkerNotRunningException()
 
         repository = get_object_or_404(GitRepository, id=pk)
-        repository.sync(user=request.user)
-        return Response({"message": f"Repository {repository} sync job added to queue."})
 
+        job_result = repository.sync(user=request.user)
+
+        return Response({
```

Comment:
Ideally we'd define a proper Serializer class for this response (and add it as the `response` type in the `extend_schema` decorator call above so that it's properly documented in Swagger). Compare for example to the `JobViewSetBase.run` view.

Reply:
Noted; I'll do this. I absolutely want to do it the "right way".

Reply:
Regarding the Swagger comment, when looking into this I notice that /`extras/git-repositories/{id}/sync/` also shows a `Request body` as `required` - which is not the case if I understand correctly - indeed I didn't think this API endpoint would do anything with a `Request body` at all.

If I'm right about that, I could try to fix that too?

Reply:
I think I'm on the right track - will follow up.

Reply:
Done; I hope I've understood correctly. See what you think.